bip78/src/receiver/mod.rs

```diff
@@ -37,28 +60,111 @@ impl UncheckedProposal {
         let reader = base64::read::DecoderReader::new(&mut limited, base64::STANDARD);
         let psbt = Psbt::consensus_decode(reader).map_err(InternalRequestError::Decode)?;
 
+        let pairs = url::form_urlencoded::parse(query.as_bytes());
+        let params = Params::from_query_pairs(pairs).map_err(InternalRequestError::SenderParams)?;
+
+        // TODO check params are valid for request Origianl PSBT
+
         Ok(UncheckedProposal {
             psbt,
+            params,
         })
     }
 
+    /// The Sender's Original PSBT
     pub fn get_transaction_to_check_broadcast(&self) -> bitcoin::Transaction {
         self.psbt.clone().extract_tx()
     }
 
-    pub fn assume_broadcastability_was_verified(self) -> UnlockedProposal {
-        UnlockedProposal {
+    /// Receiver MUST check that the Original PSBT from the sender
+    /// can be broadcast, i.e. `testmempoolaccept` bitcoind rpc returns { "allowed": true,.. }
+    /// for `get_transaction_to_check_broadcast()` before calling this method.
```

Comment:
It'd be nice to have the first line more concise, e.g. "Call after checking that the Original PSBT can be broadcast."
Longer explanation can be below.